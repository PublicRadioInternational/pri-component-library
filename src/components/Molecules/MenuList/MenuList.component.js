/**
 * @file MenuList.component.js
 * Exports a menu list component.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styles from './MenuList.css';

import List from '../../Molecules/List/List.component';

/**
 * Component that renders a Card List.
 */
const MenuList = ({ menuListItems, label }) => (
  <List
    title={label}
    className={styles.menuList}
    titleClass={styles.menuListtitle}
    liClass={styles.menuListItem}
    linkClass={styles.menuListLink}
    listItems={menuListItems}
  />
);

MenuList.propTypes = {
  menuListItems: PropTypes.arrayOf(PropTypes.object),
  label: PropTypes.string
};

MenuList.defaultProps = {
  menuListItems: [],
  label: null
};

export default MenuList;
