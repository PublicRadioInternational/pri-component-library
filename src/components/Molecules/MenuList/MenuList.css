/* import colors... */
@value colors: "../../00_global/colors.css";
@value gray, orange, blueLight from colors;

.menuList {
  margin-bottom: 1rem;
}

.menuListtitle {
  font-size: 1rem;
  font-weight: normal;
  margin-bottom: 0.5rem;
}

.menuListItem {
  display: block;
  padding: 0;
  text-align: left;
}

.menuListItem::after {
  display: none;
}

.menuListLink {
  border-left: 2px solid orange;
  color: gray;
  display: block;
  line-height: 1.5;
  padding: 3px 0 3px 10px;
}

.menuListLink:hover {
  background-color: blueLight;
  color: white;
}
