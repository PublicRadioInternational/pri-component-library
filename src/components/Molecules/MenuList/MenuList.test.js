/**
 * @file MenuList.test.js
 * Contains tests for MenuList.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import MenuList from './MenuList.component';

describe('<MenuList />', () => {
  it('Matches the Menu List snapshot', () => {
    const component = renderer.create(<MenuList />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
