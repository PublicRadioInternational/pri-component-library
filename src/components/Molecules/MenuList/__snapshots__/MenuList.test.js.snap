// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuList /> Matches the Menu List snapshot 1`] = `
<div
  aria-labelledby={null}
  className="menuList"
  id={null}
  role={null}
>
  <ul
    className="list null"
  />
</div>
`;
